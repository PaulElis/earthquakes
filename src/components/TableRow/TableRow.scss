.Row {
  display: block;
}

table {
  // font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

.table-cell {
  border: 1px solid #dddddd;
  text-align: left;
  padding: 8px;
}

tr.closed-row:nth-child(even) {
  background-color: #eee;
}

.open-cell{
  background-color: #8dd4f2;
}

.open-details-row{
  color: #0366d6;
  text-align-last: center;
  background-color: #8dd4f2;
}

tr.open-details-row td{
  border-style: hidden;
}

td.open-details-data{
  padding: 8px;
}

button{
  cursor: pointer;
}

.closed-button {
  display: flex;
  align-items: center;
  border: 1px solid #0366d6;
  border-radius: 2px;
  color: #0366d6;
  background-color: white;
}

.open-button {
  display: flex;
  align-items: center;
  color: white;
  letter-spacing: 1px;
  background-color: #0366d6;
  border: 0;
  border-radius: 2px;
}