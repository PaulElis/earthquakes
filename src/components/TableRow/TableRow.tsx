import { useState } from "react";
import Moment from "moment";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import ArrowDropUpIcon from "@material-ui/icons/ArrowDropUp";

import "./TableRow.scss";

interface Props {
  id: string;
  time: string;
  location: string;
  magnitude: number;
  latitude: number;
  longitude: number;
}

export default function TableRow({
  id,
  time,
  location,
  magnitude,
  latitude,
  longitude
}: Props) {
  const [open, setOpen] = useState(false);

  return (
    <>
      <tr className={open ? "open-row" : "closed-row"}>
        <td className="table-cell">{id}</td>
        <td className="table-cell">
          {Moment(time).format("MMMM D, YYYY @ kk:mm")}
        </td>
        <td className="table-cell">{location}</td>
        <td className="table-cell">{magnitude}</td>
        <td className={open ? "open-cell table-cell" : "table-cell"}>
          <button
            className={open ? "open-button" : "closed-button"}
            onClick={() => setOpen(!open)}
            data-testid="details-button"
          >
            Details
            {open ? <ArrowDropUpIcon /> : <ArrowDropDownIcon />}
          </button>
        </td>
      </tr>
      {open && (
        <tr className="open-details-row" data-testid="open-details-row">
          <td colSpan={6} className="open-details-data">
            Latitude {latitude} Longitude {longitude}
          </td>
        </tr>
      )}
    </>
  );
}
