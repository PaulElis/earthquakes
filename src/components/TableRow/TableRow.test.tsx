import TableRow from "./TableRow";

describe("Table Row", () => {
     test('expect TableHeader to match snapshot', () => {
          expect(TableRow).toMatchSnapshot();
     })
});
