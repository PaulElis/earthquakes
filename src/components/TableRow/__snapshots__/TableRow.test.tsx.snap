// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row expect TableHeader to match snapshot 1`] = `[Function]`;
